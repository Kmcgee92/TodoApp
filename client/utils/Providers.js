import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
// provide redux and apollo
import { Provider } from "react-redux";
import { MockedProvider } from "@apollo/client/testing";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

export default function ComponentWithProviders(
  ComponentToBeTested,
  props,
  initialState
) {
  let mocks = [];
  let store = mockStore(
    
    initialState || { auth: { activeUser: {}, error: false } }
  
  );
  return (
    <MockedProvider mocks={mocks}>
      <Provider store={store}>
        <ComponentToBeTested {...props} />
      </Provider>
    </MockedProvider>
  );
}
