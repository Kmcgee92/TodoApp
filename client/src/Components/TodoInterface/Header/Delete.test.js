import toJSON from "enzyme-to-json";

import { mount } from "enzyme";
// component to test
import Delete from "./Delete";
// child components
import DeleteIcon from "@material-ui/icons/Delete";
// utils
import Providers from "../../../../utils/Providers";

describe("<Delete/>", () => {
  let props;
  let wrapWithProviders;
  let component;
  describe("Initial Component Rendering", () => {
    beforeEach(() => {
      props = {
        classes: {},
      };
      wrapWithProviders = mount(Providers(Delete, props));
      component = wrapWithProviders.find(Delete);
    });
    it("Should Render", () => {
      expect(component).toHaveLength(1);
    });
    it("Should Match Snapshot", () => {
      expect(toJSON(component)).toMatchSnapshot();
    });
    it("Should Render NoteAddIcon", () => {
      let deleteIcon = component.find(DeleteIcon);
      expect(deleteIcon).toHaveLength(1);
      expect(deleteIcon.props().onClick).toBeDefined();
    });
  });
});

