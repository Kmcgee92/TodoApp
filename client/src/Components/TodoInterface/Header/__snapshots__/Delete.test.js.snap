// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Delete/> Initial Component Rendering Should Match Snapshot 1`] = `
<Delete
  classes={Object {}}
>
  <div>
    <ForwardRef(DeleteIcon)
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(SvgIcon))
        onClick={[Function]}
      >
        <ForwardRef(SvgIcon)
          classes={
            Object {
              "colorAction": "MuiSvgIcon-colorAction",
              "colorDisabled": "MuiSvgIcon-colorDisabled",
              "colorError": "MuiSvgIcon-colorError",
              "colorPrimary": "MuiSvgIcon-colorPrimary",
              "colorSecondary": "MuiSvgIcon-colorSecondary",
              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
              "root": "MuiSvgIcon-root",
            }
          }
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </ForwardRef(SvgIcon)>
      </WithStyles(ForwardRef(SvgIcon))>
    </ForwardRef(DeleteIcon)>
  </div>
</Delete>
`;
